einops/_backends.py

```diff
@@ -662,3 +662,58 @@ def is_float_type(self, x):
 
     def einsum(self, pattern, *x):
         return self.tinygrad.Tensor.einsum(pattern, *x)
+
+
+class PyTensorBackend(AbstractBackend):
+    framework_name = "pytensor"
+
+    def __init__(self):
+        from pytensor import tensor
+
+        self.pt = tensor
+
+    def is_appropriate_type(self, tensor):
+        return isinstance(tensor, self.pt.TensorVariable)
+
+    def is_float_type(self, x):
+        return x.dtype in self.pt.type.float_dtypes
+
+    def from_numpy(self, x):
+        return self.pt.as_tensor(x)
+
+    def to_numpy(self, x):
+        return x.eval()  # Will only work if there are no symbolic inputs
+
+    def create_symbol(self, shape):
+        if not isinstance(shape, tuple | list):
+            shape = (shape,)
+        return self.pt.tensor(shape=shape)
+
+    def eval_symbol(self, symbol, input_dict):
+        # input_dict is actually a list of tuple?
+        return symbol.eval(dict(input_dict))
+
+    def arange(self, start, stop):
+        return self.pt.arange(start, stop)
+
+    def shape(self, x):
+        # use the static shape dimensions where known
+        return tuple(
+            static_dim if static_dim is not None else symbolic_dim
+            for static_dim, symbolic_dim in zip(x.type.shape, x.shape)
+        )
```

Comment:
Could you give me a brief summary of how `shape` is used internally? I could just return x.shape if it has no performance penalty

Reply:
if it is static, then divisibility checks will report problems earlier (which is likely helpful in theano/pytensor).

Reply:
Great, will leave it like this then